cmd/go-clang-generate/main.go

```diff
@@ -331,39 +383,62 @@ func main() {
 	for _, f := range functions {
 		fname := f.Name
 
+		// Prepare the parameters
 		for i := range f.Parameters {
 			p := &f.Parameters[i]
 
-			if n, ok := lookupNonTypedefs[p.Type.Name]; ok {
+			if n, ok := lookupNonTypedefs[p.Type.CName]; ok {
 				p.Type.Name = n
 			}
 			if e, ok := lookupEnum[p.Type.Name]; ok {
 				p.CName = e.Receiver.CName
-				p.Type = e.Receiver.Type
+				// TODO remove the receiver... and copy only names here to preserve the original pointers and so
+				p.Type.Name = e.Receiver.Type.Name
+				p.Type.CName = e.Receiver.Type.CName
+				p.Type.Primitive = e.Receiver.Type.Primitive
 			} else if _, ok := lookupStruct[p.Type.Name]; ok {
-			} else {
-				if goType, primitiveType := goAndTypePrimitive(p.Type.Name); goType != "" {
-					p.Type.Name = goType
-					p.Type.Primitive = primitiveType
+			}
+
+			// TODO happy hack, whiteflag types that are return arguments
+			if p.Type.PointerLevel == 1 && (p.Type.Name == "File" || p.Type.Name == "FileUniqueID" || p.Type.Name == "IdxClientFile" || p.Type.Name == "cxstring" || p.Type.Name == GoUInt16) {
```

Comment:
possibly define a whitelist map? would get rid of this || p.Type.Name == cascade